src/transformers/adapters/heads.py

```diff
@@ -520,6 +520,21 @@ def add_prediction_head(
                 f"Model already contains a head with name '{head.name}'. Use overwrite_ok=True to force overwrite."
             )
 
+    def delete_head(self, head_name: str):
+        """
+        Deletes the prediction head with the specified name from the model.
+
+        Args:
+            head_name (str): The name of the prediction to delete.
+        """
+        if head_name not in self.config.prediction_heads:
+            logger.info("No prediction head '%s' found for deletion. Skipping.", head_name)
+            return
+        del self.config.prediction_heads[head_name]
+        del self.heads[head_name]
+        if self.active_head == head_name:
+            self.active_head = None
+
```

Comment:
When deleting a head that is the active head, the active head is set to None but when deleting an adapter in the active adapters setup the setup remains as it is. Maybe it would be better to handle these two cases similarily